gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/filter/RankingFilter.java

```diff
@@ -0,0 +1,199 @@
+            @JsonProperty("attributes") final List<Qualifier> attributes,
+            @JsonProperty("operator") final String operator,
+            @JsonProperty("value") final Integer value) {
+        this.measures = notNull(measures, "measures must not be null!");
+        this.attributes = attributes;
+        this.operator = notNull(operator, "operator must not be null!");
+        this.value = notNull(value, "value must not be null!");
+    }
+
+    public RankingFilter(
+            final List<Qualifier> measures,
+            final List<Qualifier> attributes,
+            final RankingFilterOperator operator,
+            final Integer value) {
+        this(measures, attributes, notNull(operator, "operator must not be null!").name(), value);
+    }
+
+    /**
+     * Returns all the qualifiers used by the ranking filter.
+     * <p>
+     * This information comes handy if it is necessary, for example, to convert the ranking filter to use just
+     * the URI object qualifiers instead of the identifier object qualifiers. It can be used to gather these
+     * for a conversion service.
+     *
+     * @return all the qualifiers the ranking filter uses
+     */
+    @JsonIgnore
+    public Collection<ObjQualifier> getObjQualifiers() {
+        return Stream.concat(
+                this.measures.stream(),
+                this.attributes == null ? Stream.empty() : this.attributes.stream()
+        )
+                .filter(ObjQualifier.class::isInstance)
+                .map(ObjQualifier.class::cast)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Copy itself using the given object qualifier converter in case when {@link IdentifierObjQualifier} instances are used in the object otherwise the
+     * original object is returned.
+     * <p>
+     * The provided converter must be able to handle the conversion for the qualifiers that are of the {@link IdentifierObjQualifier} type that are used by
+     * this object or its encapsulated child objects.
+     *
+     * @param objQualifierConverter The function that converts identifier qualifiers to the matching URI qualifiers. In case when the object uses the
+     *         identifier qualifiers, it
+     *         will return a new copy of itself or its encapsulated objects that used URI qualifiers, otherwise the original object is returned.
+     *         The parameter must not be null.
+     *
+     * @return copy of itself with replaced qualifiers in case when some {@link IdentifierObjQualifier} were used, otherwise original object is returned.
+     *
+     * @throws IllegalArgumentException The exception is thrown when conversion for the identifier qualifier used by this ranking filter could not be
+     *         made by the provided
+     *         converter or when provided converter is null.
+     */
+    public RankingFilter withObjUriQualifiers(final ObjQualifierConverter objQualifierConverter) {
```

Comment:
I'm not sure about the use-case for this method, but it is very similar to [MeasureDefinition#withObjUriQualifiers](https://github.com/gooddata/gooddata-java/blob/da16ae63b448b40fadf08c0ecfd4f71f87d30bbc/gooddata-java-model/src/main/java/com/gooddata/sdk/model/executeafm/afm/MeasureDefinition.java#L73). We should consider to extract some interface for this in the future.

Reply:
AFM supports identifier and uri qualifiers of metadata objects but old ReportDefinition executor supports only uri. Webapp needs some copy method for AFM sub-objects to recreate them with uri qualifiers only (after it extracts them and then translates them). Maybe we could make them more generic in future.